yourgov/components/FormMobileFoodVendorPermit/steps/FormStep9.tsx

```diff
@@ -1,72 +1,292 @@
+		);
+		setCurrentDocument(document);
+		closeDrawerAndClearForm();
+	};
 
-	const typeCorrectedErrors = errors as ShallowErrors<Step9FormSchema>;
+	const onSaveAndContinue = () => {
+		const documentsWithErrors = documents.map((document) =>
+			document.file
+				? document
+				: {
+						...document,
+						error: true,
+				  }
+		);
+		const tableErrors = documentsWithErrors.reduce<SampleTableError[]>(
+			(acc, document) =>
+				document.file
+					? acc
+					: [
+							...acc,
+							{
+								documentType: document.documentType,
+								href: `#${document.id}`,
+								message: `Upload a file for ${document.documentType}`,
+								name: 'files',
+							},
+					  ],
+			[]
+		);
 
-	const onSubmit: SubmitHandler<Step9FormSchema> = async (data) => {
+		setCurrentDocument(undefined);
+		setDocuments(documentsWithErrors);
+		setTableErrors(tableErrors);
+
+		return {
+			hasErrors: documentsWithErrors.some(({ error }) => error),
+		};
+	};
+
+	const onSubmit: FormEventHandler<HTMLFormElement> = async (event) => {
+		event?.preventDefault();
 		if (isSavingBeforeExiting) {
 			return;
 		}
+		const { hasErrors } = onSaveAndContinue();
+		if (hasErrors) return;
+
 		await submitStep();
+
 		step9SetState({
-			...data,
+			files: Object.fromEntries(
+				documents.map(({ file, id, size }) => [id, { file, size }])
+			) as Record<Document['id'], { file: string; size: string }>,
 			completed: !isSavingBeforeExiting,
-			started: true,
 		});
 	};
 
+	const isSuccessMessageVisible = currentDocument && !isDrawerOpen;
+
 	return (
 		<FormContainer
 			formTitle="Upload documents"
 			formIntroduction="Upload all documents listed in the table below."
 		>
-			<Stack as="form" gap={3} onSubmit={handleSubmit(onSubmit)} noValidate>
-				<Controller
-					control={control}
-					name="cuisine"
-					render={({ field: { ref, ...field } }) => (
-						<ComboboxMulti
-							label="TODO"
-							hint="TODO"
-							inputRef={ref}
-							options={cuisineOptions}
-							required
-							{...field}
-							id="cuisine"
-							invalid={Boolean(typeCorrectedErrors.cuisine?.message)}
-							message={typeCorrectedErrors.cuisine?.message}
-							maxWidth="xl"
-							block={false}
+			<Stack gap={2}>
+				<Stack gap={1}>
+					<Box css={tableErrors.length ? undefined : { display: 'none' }}>
+						<SectionAlert
+							focusOnUpdate={tableErrors}
+							ref={errorMessageRef}
+							tabIndex={-1}
+							title="You must provide all documents in the table below"
+							tone="error"
+						>
+							<Stack gap={0.5} alignItems="flex-start">
+								<UnorderedList>
+									{tableErrors.map(({ href, message }) => (
+										<ListItem key={href}>
+											<TextLink href={href}>{message}</TextLink>
+										</ListItem>
+									))}
+								</UnorderedList>
+							</Stack>
+						</SectionAlert>
+					</Box>
+
+					<Box css={isSuccessMessageVisible ? undefined : { display: 'none' }}>
+						<SectionAlert
+							focusOnUpdate={documents}
```

Comment:
Annoyingly, VoiceOver doesn't seem to announce this when it's focused 😭

Reply:
It seems to focus div, then the upload button(?) then the div and perhaps that's why it's not announced? 
<img width="947" alt="image" src="https://github.com/user-attachments/assets/b4a6e368-a31f-455a-a235-db59f24ff5d9">

Reply:
- [ ] look into removing `focusOnUpdate`